### ISSUE TITLE
Problems occurred in ToF extraction after CSI extraction

### ISSUE THREAD

**Issue by crownz-sec** [IMDEANetworksWNG/UbiLocate#22]
At first, I extracted CSI using `bash reload.sh` `bash config.sh nss ch bw` `bash send_collect.sh name ss packets bw`. There is no errors. After that, I began to extract ToF. When I execute this command `bash config.sh`,I get the following error output in the terminal:
```
Reloading the devices
/usr/bin:/bin
/bin:/usr/bin:/sbin:/usr/sbin:/home/imdea:/mmc/sbin:/mmc/bin:/mmc/usr/sbin:/mmc/usr/bin:/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin
Chanspec set to 0xe29b
2
Read 476 bytes
1+0 records in
1+0 records out
4 bytes (4B) copied, 0.000054 seconds, 72.3KB/s
__nex_driver_io: error
wl: Unsupported
wl: Unsupported
__nex_driver_io: error
wl: Unsupported
wl: Unsupported
/usr/bin:/bin
/bin:/usr/bin:/sbin:/usr/sbin:/home/imdea:/mmc/sbin:/mmc/bin:/mmc/usr/sbin:/mmc/usr/bin:/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin
Chanspec set to 0xe29b
3
Read 476 bytes
1+0 records in
1+0 records out
4 bytes (4B) copied, 0.000049 seconds, 79.7KB/s
__nex_driver_io: error
wl: Unsupported
wl: Unsupported
__nex_driver_io: error
wl: Unsupported
wl: Unsupported
```
What's wrong with it? Do I need to restart the router if I want to extract ToF after extracting CSI?

**Comment by alejandroBlancoPizarro** [comment]
Hi,

It sounds that you did not successfully load the dhd.ko module. Check the output of md5sum of dhd.ko inside the router and compare it with the one in the GitHub repository in tofsoftware.

Best,
Alejandro

**Comment by crownz-sec** [comment]
I have already check the md5sum on all routers, which is same with the file inside the repository,and value is: e3b655e68967b60edb922f65662cd8a8.

I tried rebooting all the routers, and I found that this would execute the bash file of ToF without error. Can I think it is necessary to restart routers to switch the extraction CSI or ToF?

**Comment by alejandroBlancoPizarro** [comment]
Then, I would say that you did not reload.sh succesfully. Try to execute on the router the commands inside these files:reload.sh  config.sh nss ch bw

After that, let me know

**Comment by crownz-sec** [comment]
> Then, I would say that you did not reload.sh succesfully. Try to execute on the router the commands inside these files:reload.sh config.sh nss ch bw
> 
> After that, let me know

In order to make sure that my problems are repeatable. I tried the same operation for four more times, extracting CSI first and then ToF. When extracting CSI, there is no problem.  But when extracting ToF, I found that the execution of script would be blocked for a short time(both `reload.sh` and `config.sh`).And I even use ctrl+c to interrupt the excution of `config.sh`, after that, I executed `config.sh` again, I do not know whether the execution was successful, but according to the output of config.sh, it should have failed.(The output is same with above I mentioned.)

At the same time, I noticed that when I executed the script of extracting CSI, I could not search the WiFi network. When I executed the config scripts of ToF for several times, I could search the WiFi network again. At this time, I executed the bash scripts of ToF and found that .pcap files could be extracted. 

Finally, I tried what you said. Similarly, ToF was extracted after CSI extraction, but this time ToF config bash files were executed on the router(I executed `config.sh` `reload.sh` and `./cmdserver -p 30000` on the router.). After the execution, ToF extraction command was executed on my own computer, no problem happens and I can get .pcap files.

I can't pinpoint the problem, I can only describe to you what I have tried.
